src/query_plan/operation.rs

```diff
@@ -601,8 +571,19 @@ impl NormalizedSelection {
         }
     }
 
-    pub(crate) fn has_defer(&self) -> Result<bool, FederationError> {
-        todo!()
+    pub(crate) fn has_defer(&self) -> bool {
```

Comment:
I am not sure if the return type of the stub - `Result<bool, FederationError>` - was intentional. The JS implementation just returns a bool, and I am not sure in which cases we'd want to throw an error. Let me know if the `Result` was there for a reason.

Reply:
@SimonSapin iirc this was your original stub. did you have a plan for the `Result`?

Reply:
I don't see a case where this could potentially throw. Lets keep it as `bool` and change it later on if needed.

Reply:
I believe attempting to specify `@defer` on a field selection could result in an error? Since parsed document is validated I guess maybe this was a safeguard so we don't transform selection set to something invalid? I guess if you incorrectly try to normalize inline fragment spread to lift up selection you could end up with something invalid.... but that would be a bug in the code?

Reply:
JS code doesn't throw so we should keep it as `bool`